components/homme/src/test_src/dry_planar.F90

```diff
@@ -283,6 +282,10 @@ subroutine bubble_init(elem,hybrid,hvcoord,nets,nete,f)
 
   hvcoord%hyai = ai; hvcoord%hybi = bi
 
+  do k = 1,nlev
```

Comment:
why this change?

Reply:
Previously, it was computing it according to the analytical formula. But that is not BFB in finite precision with the pressure_thickness call, which is the normal way to init dp3d in standalone tests. This was causing non-BFBness in the C++-vs-F90 test for the bubble tests, since Hommexx's initialization assumes that dp3d is computed w.r.t. ps by the usual formula. Differences in dpm in bubble_init are at machine precision.

Reply:
Ok, my first thought was to revert it. I do not believe we should stick to dcmip wrappers code. However if that it too much to change xx init, then it is ok to keep it. i think xx init should have mimic-ed F init though (i assume it is about routine set_elem_state() ).

Reply:
The error, in my view, is in bubble_init. It sets up ps and dp3d in such a way that even if nothing happened in the first time step, dp3d would change just due to using the standard formula in, e.g., vertical remap rather than bubble_init's computation. In contrast, all the other standalone tests use pressure_thickness for finite-precision consistency.

Reply:
There is no reason to use A, B in dp3d away from init, and in init it is only a convenience. I do not see it as an error, however, it is a bigger change to fix this in xx, so let's keep this change.

Reply:
Just for clarity, by consistency I mean e.g. with:
https://github.com/E3SM-Project/E3SM/blob/c7b8bd8d81c4f8f22c5686dc0a17c37c68ef95d1/components/homme/src/theta-l/share/vertremap_mod.F90#L81-L82
If we want to remove a and b, it's not just an init-time issue or an Hxx issue.

Reply:
I understand. At this point it is just a philosophical conversation about hybrid coefs, which we can postpone.

Reply:
To ddcument this further, in Chrysalis, homme_integration shows diffs for the F90 bubble tests at machine precision level, consistent with the machine-precision change to dpm in bubble_init:
```
############################################################################
CPRNC returned the following RMS differences
 RMS T                                7.0545E-13            NORMALIZED  2.4100E-15
 RMS Th                               4.4738E-14            NORMALIZED  1.4912E-16
 RMS ps                               9.2247E-11            NORMALIZED  9.2247E-16
 RMS T                                8.1605E-12            NORMALIZED  2.7882E-14
 RMS Th                               6.1777E-12            NORMALIZED  2.0591E-14
 RMS geo                              4.3877E-11            NORMALIZED  5.9630E-15
############################################################################
```